### ISSUE TITLE
Game scoring error?

### ISSUE THREAD

**Issue by pheasant75** [gcp/leela-zero#423]
I was watching this match game and it showed W+4.5. 
http://zero.sjeng.org/viewmatch/178a8461a877ae2e4e12db74f90d2580da874a1a5e7ae0abf0f94767704c1239
However, I counted by hand and white have 173 territory.  So White lost.  What's going on here?

**Comment by pheasant75** [comment]
Also this game:
http://zero.sjeng.org/viewmatch/dfd2089da67d69fdf4202fa698067eae0418de778afe4df302ccff8435294ee4?viewer=wgo
The result showed w win by 1.5.  However, if we click those dead stones/groups to make them dead (especially the black group at bottom), white wins by 31.5.

**Comment by roy7** [comment]
That isn't how Tromp Taylor scoring works though. All stones are always alive. LZ needs to capture them if she thinks they are dead.

**Comment by roy7** [comment]
What makes things even more annoying is WGO only does, I think, Japanese scoring, not Chinese. So it's even more of than usual. (Removing stones didn't affect the score in WGo, but Chinese gets 1 point per stone.)

**Comment by pheasant75** [comment]
But how the Tromp Taylor compute territory around the "dead" stone if it is not being captured?  Shared?  
I worry that the imprecise score method will lead to bias of the models.  It makes the model mis-judge moves.  It is like providing a different go rule to LZ.  
In my first example, black wins, and he knows it because he passed many times near the end.  Yet the system think white wins and possibly choose a weaker model.  
Is there better scoring method?

**Comment by WhiteHalmos** [comment]
Black "thinks" it has won. Now this game will show black that it needs to actually take out those white stones. If enough of these examples accumulate, training will eventually cause black to actually remove those white stones before it thinks it has won.

**Comment by WhiteHalmos** [comment]
Tromp Taylor scoring **is** the rule set that LZ is trained on, no other rule sets. Other scoring rules like Chinese or Japanese are for humans. It so happens that if a bot does well with Tromp Taylor scoring, it will also do well with most other rule sets.

**Comment by pheasant75** [comment]
ok, so if both side always use the same algorithm to evaluate the situation, the black knows that it is losing and should play instead of passing to increase the chance of winning.  So I am curious why black still passes.

**Comment by grolich** [comment]
Also, Tromp Taylor scoring is the same scoring method used in AlphaGo and in many others as well.

It removes any ambiguity from the situation (regarding misevaluating the live/dead stones). 
At first the bot doesn't know it could have won by removing the stones or that it loses without it.
After a while, the bot learns when it has to take action and remove the stones.
After a longer while, if it gets better, it starts knowing that it lost without having to wait for these stones to be removed (it knows the opponent can remove them and win).

The other way around - when it is winning, is a different issue, but doesn't have any impact on the result, since Tromp Taylor scoring is an area scoring variant, where your stones are counted as well, so even in these cases, it will lose no points.

In addition, there is no "precise" scoring method when games are undecided (areas not fully enclosed), and so Tromp Taylor can be used for many situations which don't normally arise in human play as well as those that do.

**Comment by grolich** [comment]
As to why black passes here - there are several possibilities. The possibility of a bug is always there, of course, but there are simpler ones.

For example -The neural network gets more than just the board state. It also gets other data, such as if the side to move is white or not, so it could also be that the Neural Network still hasn't fully learned how to factor in the komi in all positions, making black more likely to think he won in some positions where he hasn't. 
Makes sense, as it has far more serious problems to fix in its play yet.

As for why the MCTS doesn't reduce the odds of passing sufficiently as it sees it loses? Well, it might not see that at all. It has to look at a black pass followed by a white pass and if the initial policy gives passing a high probability, it might not have learned enough to reduce it sufficiently.

In other words: Leela still has much to learn.
Personally, I hope it's such a simple explanation and not another bug, but it doesn't hurt to keep our eyes peeled for those as well.
